### DISCUSSION TITLE
Cannot move on from Step 4 in skills-with-codespaces

### DISCUSSION THREAD

**Discussion by YeoYiXin** [skills/.github#384] [Introduction to GitHub] [answered]
Hello, I am in Step 4 of Code-with-Codespaces. I have done according to the steps and I managed to get the train animation in my terminal. After waiting for 20s, I refreshed the page but found out that I am unable to move on. Hence, I look into why I couldn't move on. It turns out that some jobs were unsuccessful. The error is Failed on update setup. Hence, I tried to redo the steps and now, I couldn't manage to get the train animation after reopening codespace even after following all the steps. I already installed sl but it said sl not found. However, if I continued the steps until /users/games/sl right after installing everything (without closing codespace tab), it works but I cannot move on due to the error. I tried deleting the whole repository and redo it multiple times but I am still stuck at Step 4. I would really appreciate if someone can help me out! Thank you in advance!

**Comment by kazzyfrog** [comment, marked as answer]
Hello @YeoYiXin !
Have you created a setup.sh file and added the following code within that file?
```
#!/bin/bash

sudo apt-get update
sudo apt-get install sl
```
I have completed this course. Please check it out.  https://github.com/kazzyfrog/skills-code-with-codespaces
If you have any other questions, I will help.

**Reply by YeoYiXin** [reply]
It works now, thank you so much!!

**Reply by kazzyfrog** [reply]
I'm glad I could help out.
